haystack/document_stores/opensearch.py

```diff
@@ -600,47 +508,223 @@ def _create_document_index(self, index_name: str, headers: Optional[Dict[str, st
                     index_definition["mappings"]["properties"].update({field: {"type": "text"}})
 
             if self.embedding_field:
-                index_definition["settings"]["index"] = {"knn": True}
+                index_definition["settings"]["index"] = {"knn": True}  # TODO: option to turn off for script scoring
```

Comment:
We would need to:
- set it here accordingly
- add an additional validation step to check whether knn is set to true when using ANN.
- add tests
It's not huge to add this, but I felt that this PR is already quite big. So I would leave that for an additional PR. But happy to implement it here if you think otherwise.

Reply:
Sure, let's keep this for another PR.